### PR TITLE
67 fix bug deployment with end date

### PR DESCRIPTION
**Pull request by andriwild**
- Added the ability to create deployments with an end date in the future.
- Updated deployment validation endpoint to handle date ranges without a defined end date. This is one possible solution to this problem, we probably need to look at other ways?

### CODE REVIEW

File: services/api/routers/validators.py [hunk 1]
```diff
@@ -16,13 +16,14 @@
...
 async def validate_deployment(body: DeploymentRequest) -> None:
+
     transaction = await database.transaction()
     try:
 
         values = {
             deployments.c.node_id: body.node_id,
             deployments.c.location: text('point(:lat,:lon)').bindparams(lat=body.location.lat,lon=body.location.lon),
-            deployments.c.period: to_inclusive_range(body.period),
+            deployments.c.period: to_inclusive_range(body.period) if body.period != None else None,
```
Comment (cspindler) [inline services/api/routers/validators.py]: Deployments without period are not valid. The end part of a period can be open, though. [`to_inclusive_range`](https://github.com/mitwelten/mitwelten-db-backend/blob/97d23e5434d2250cfbf12d06ffaa92777c23457a/services/api/dependencies.py#L134) can deal with an open range. Did the validator cause problems during testing?

**Review by cspindler — CHANGES_REQUESTED** [review]
Check the validator, I'd like to have the deployment period mandatory.

File: services/api/routers/validators.py (hunk 1 above)
Comment (andriwild) [reply services/api/routers/validators.py]: It throws an exception if the `to_inclusive_range` function is called with None, which happens quite often when no start and end dates are set.

Comment (cspindler) [reply services/api/routers/validators.py]: I fixed the validator in `TimeStampRange`.
This also fixes the "None" problem of `to_inclusive_range`.
Both `start` and `end` are now optional, to allow deployments that "always exist".

**Review by cspindler — APPROVED** [review]

**Pull request merged**